src/test/java/io/aiven/kafka/connect/common/output/avro/AvroOutputWriterTest.java

```diff
@@ -0,0 +1,213 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.common.output.avro;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.kafka.common.record.TimestampType;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.header.ConnectHeaders;
+import org.apache.kafka.connect.header.Headers;
+import org.apache.kafka.connect.sink.SinkRecord;
+
+import io.aiven.kafka.connect.common.config.OutputField;
+import io.aiven.kafka.connect.common.config.OutputFieldEncodingType;
+import io.aiven.kafka.connect.common.config.OutputFieldType;
+
+import org.apache.avro.file.DataFileReader;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.util.Utf8;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class AvroOutputWriterTest {
+
+    private @TempDir Path tmpDir;
+
+    @Test
+    void testWriteAllFields() throws IOException {
+        final Path avroFile = tmpDir.resolve("avro.file");
+        final List<String> values = List.of("a", "b", "c", "d");
+        writeRecords(
+            avroFile,
+            List.of(
+                new OutputField(OutputFieldType.KEY, OutputFieldEncodingType.NONE),
+                new OutputField(OutputFieldType.OFFSET, OutputFieldEncodingType.NONE),
+                new OutputField(OutputFieldType.TIMESTAMP, OutputFieldEncodingType.NONE),
+                new OutputField(OutputFieldType.HEADERS, OutputFieldEncodingType.NONE),
+                new OutputField(OutputFieldType.VALUE, OutputFieldEncodingType.NONE)
+            ),
+            SchemaBuilder.STRING_SCHEMA,
+            values,
+            true,
+            true
+        );
+        int counter = 0;
+        final int timestamp = 1000;
+        final Map<Utf8, ByteBuffer> expectedHeaders = new HashMap<>();
+        expectedHeaders.put(new Utf8("a"), ByteBuffer.wrap("b".getBytes(StandardCharsets.UTF_8)));
+        expectedHeaders.put(new Utf8("c"), ByteBuffer.wrap("d".getBytes(StandardCharsets.UTF_8)));
+        for (final GenericRecord r : readRecords(avroFile)) {
+            final GenericData.Record expectedRecord = new GenericData.Record(r.getSchema());
+            expectedRecord.put("key", new Utf8("some-key-" + counter));
+            expectedRecord.put("offset", 100L);
```

Comment:
Can we avoid hardcoding these values in 2 places? Otherwise it's a little bit unclear where they are coming from unless looking into `writeRecords`